### PR TITLE
Add `getHashRate` endpoint

### PR DESCRIPTION
**Pull request by Lbqds**
solve #485

### CODE REVIEW

File: app/src/main/scala/org/alephium/app/ServerUtils.scala [hunk 1]
```diff
@@ -64,6 +65,25 @@ class ServerUtils(implicit
...
+      case Right(blocks) =>
+        val hashCount = blocks.fold(BigInt(0)) { case (acc, entries) =>
+          entries.fold(acc) { case (hashCount, entry) =>
+            val target   = entry._1.target
+            val hashDone = Target.maxBigInt.divide(target.value)
+            hashCount + hashDone
+          }
+        }
+        val timeSpan = toTs.deltaUnsafe(fromTs)
+        val hashrate = (hashCount * 1000 * groupConfig.chainNum) / timeSpan.millis
```
Comment (polarker) [inline app/src/main/scala/org/alephium/app/ServerUtils.scala:81]: We could use this function to calculate hashrate: https://github.com/alephium/alephium/blob/master/flow/src/main/scala/org/alephium/flow/handler/ChainHandler.scala#L220-L221

Comment (polarker) [reply app/src/main/scala/org/alephium/app/ServerUtils.scala:81]: As discussed, we will not use this method

**Comment by polarker** [comment]
@Lbqds I added a new commit with minor updates, please help me review

File: api/src/main/scala/org/alephium/api/model/TimeSpan.scala [hunk 2]
```diff
@@ -16,4 +16,8 @@
 
 package org.alephium.api.model
 
-final case class TimeSpan(seconds: Int) extends AnyVal
+import org.alephium.util.Duration
+
+final case class TimeSpan(millis: Long) extends AnyVal {
```
Comment (polarker) [inline api/src/main/scala/org/alephium/api/model/TimeSpan.scala:21]: using `millis` is consistent with other endpoints

File: api/src/main/scala/org/alephium/api/model/TimeInterval.scala [hunk 3]
```diff
@@ -16,6 +16,19 @@
...
+final case class TimeInterval(from: TimeStamp, to: TimeStamp) {
+  def validateTimeSpan(max: Duration): Either[ApiError[_ <: StatusCode], Unit] = {
+    if (durationUnsafe() > max) {
+      Left(ApiError.BadRequest(s"Time span cannot be greater than ${max}"))
+    } else {
+      Right(())
+    }
+  }
+
+  def durationUnsafe(): Duration = to.deltaUnsafe(from)
```
Comment (Lbqds) [inline api/src/main/scala/org/alephium/api/model/TimeInterval.scala:33]: we need to make sure `durationUnsafe() > 0`, maybe we can modify the condition to `timeInterval.fromTs >= timeInterval.toTs` at [timeIntervalQuery](https://github.com/alephium/alephium/blob/e69b108e6028ea860afc7e73bcae3bc2efd21a08/api/src/main/scala/org/alephium/api/Endpoints.scala#L56)?

Comment (polarker) [reply api/src/main/scala/org/alephium/api/model/TimeInterval.scala:33]: Nice catch! I will fix it

**Pull request merged**